test/models/parser/twitter_profile_test.rb

```diff
@@ -71,62 +60,123 @@ def empty_doc
-    assert_match /pbs.twimg.com\/profile_images\/877631054525472768\/Xp5FAPD5.jpg/, data['author_picture']
+    stub_profile_lookup.returns(twitter_profile_response_success)
+    
+    data = Parser::TwitterProfile.new('https://www.twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_equal 'fake_user', data['external_id']
+    assert_equal '@fake_user', data['username']
+    assert_match /The world's most powerful space telescope/, data['description']
+    assert_match 'fake_user', data['title']
+    assert_match 'Fake User', data['author_name']
+    assert_match 'https://twitter.com/fake_user', data['url']
+    assert_match /pbs.twimg.com\/profile_images\/685182791496134658\/Wmyak8D6.jpg/, data['picture']
+    assert_match /pbs.twimg.com\/profile_images\/685182791496134658\/Wmyak8D6.jpg/, data['author_picture']
     assert_not_nil data['published_at']
-
     assert_nil data['error']
+  end  
+  
+  test "should store raw data of profile returned by Twitter API" do
+    stub_profile_lookup.returns(twitter_profile_response_success)
+    
+    data = Parser::TwitterProfile.new('https://www.twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_not_nil data['raw']['api']
+    assert !data['raw']['api'].empty?
+  end  
+  
+  test "returns data even if an error is returned" do
+    stub_profile_lookup.returns(twitter_profile_response_error)
+
+    data = Parser::TwitterProfile.new('https://twitter.com/fake_user').parse_data(empty_doc)
+    
+    assert_not_nil data['error']
+    assert_equal 'fake_user', data['external_id']
+    assert_equal 'https://twitter.com/fake_user', data['url']
   end
 
-  test "should store raw data of profile returned by Twitter API" do
-    Twitter::REST::Client.any_instance.stubs(:user).returns(fake_twitter_user)
+  test "should remove line breaks from Twitter profile description" do
+    stub_profile_lookup.returns(twitter_profile_response_success)
 
-    data = Parser::TwitterProfile.new('https://www.twitter.com/fakeaccount').parse_data(empty_doc)
+    data = Parser::TwitterProfile.new('https://twitter.com/fake_user').parse_data(empty_doc)
 
-    assert_not_nil data['raw']['api']
-    assert !data['raw']['api'].empty?
+    assert_match "Launched: Dec. 25, 2021. First images revealed: July 12, 2022. Verification: https://t.co/ChOEslj1j5", data['description']
   end
 
-  test "should throw Pender::Exception::ApiLimitReached when Twitter::Error::TooManyRequests is thrown" do
-    Twitter::REST::Client.any_instance.stubs(:user).raises(Twitter::Error::TooManyRequests)
+  test "should parse tweet url with special chars, and strip them" do
+    stub_profile_lookup.returns(twitter_profile_response_success)
 
-    assert_raises Pender::Exception::ApiLimitReached do
-      Parser::TwitterProfile.new('https://twitter.com/fake-account').parse_data(empty_doc)
-    end
+    parser = Parser::TwitterProfile.new('https://0.twitter.com/fake_user')
+    data = parser.parse_data(empty_doc)
```

Comment:
small thing, but I think it helps with understandability - we don't need to assign the data variable since we don't use its result. can just do `parser.parse_data(empty_doc)` to get the side effects, but still test parser